src/Test/UnitTests/constraints.jl

```diff
@@ -22,3 +22,143 @@ function getconstraint(model::MOI.ModelLike, config::TestConfig)
+            for (index, solution_value) in constraint_dual
+                @test MOI.canget(model, MOI.ConstraintDual(), typeof(index))
+                @test MOI.get(model, MOI.ConstraintDual(), index) ≈ solution_value atol=atol rtol=rtol
+            end
+        end
+    end
+end
+
+"""
+    Add an ScalarAffineFunction-in-LessThan constraint.
+    If `config.solve`, solve and test solution.
+    In addition, if `config.duals`, test dual solution.
+"""
+function solve_affine_lessthan(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    MOIU.loadfromstring!(model,"""
+        variables: x
+        maxobjective: 1.0x
+        c: 2.0x <= 1.0
+    """)
+    x = MOI.get(model, MOI.VariableIndex, "x")
+    c = MOI.get(model, MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}}, "c")
+    if config.solve
+        testsolution(model, config;
+            objective_value   = 0.5,
+            variable_primal   = [(x, 0.5)],
+            constraint_primal = [(c, 1.0)],
+            constraint_dual   = [(c, -0.5)]
+        )
+    end
+end
+unittests["solve_affine_lessthan"] = solve_affine_lessthan
+
+"""
+    Add an ScalarAffineFunction-in-GreaterThan constraint.
+    If `config.solve`, solve and test solution.
+    In addition, if `config.duals`, test dual solution.
+"""
+function solve_affine_greaterthan(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    MOIU.loadfromstring!(model,"""
+        variables: x
+        minobjective: 1.0x
+        c: 2.0x >= 1.0
+    """)
+    x = MOI.get(model, MOI.VariableIndex, "x")
+    c = MOI.get(model, MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}}, "c")
+    if config.solve
+        testsolution(model, config;
+            objective_value   = 0.5,
+            variable_primal   = [(x, 0.5)],
+            constraint_primal = [(c, 1.0)],
+            constraint_dual   = [(c, 0.5)]
+        )
+    end
+end
+unittests["solve_affine_greaterthan"] = solve_affine_greaterthan
+
+"""
+    Add an ScalarAffineFunction-in-EqualTo constraint.
+    If `config.solve`, solve and test solution.
+    In addition, if `config.duals`, test dual solution.
+"""
+function solve_affine_equalto(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    MOIU.loadfromstring!(model,"""
+        variables: x
+        minobjective: 1.0x
+        c: 2.0x == 1.0
+    """)
+    x = MOI.get(model, MOI.VariableIndex, "x")
+    c = MOI.get(model, MOI.ConstraintIndex{MOI.ScalarAffineFunction{Float64}, MOI.EqualTo{Float64}}, "c")
+    if config.solve
+        testsolution(model, config;
+            objective_value   = 0.5,
+            variable_primal   = [(x, 0.5)],
+            constraint_primal = [(c, 1.0)],
+            constraint_dual   = [(c, 0.5)]
+        )
+    end
+end
+unittests["solve_affine_equalto"] = solve_affine_equalto
+
+"""
+    Add an ScalarAffineFunction-in-Interval constraint.
```

Comment:
Why is it indented ? Why don't you have `solve_affine_interval(...)` indented and then the docstring unindented ?